go/enclave/components/rollup_consumer.go

```diff
@@ -111,30 +118,53 @@ func (rc *rollupConsumerImpl) getSignedRollup(rollups []*common.ExtRollup) ([]*c
 
 // todo - when processing the rollup, instead of looking up batches one by one, compare the last sequence number from the db with the ones in the rollup
 // extractRollups - returns a list of the rollups published in this block
-func (rc *rollupConsumerImpl) extractRollups(br *common.BlockAndReceipts) []*common.ExtRollup {
+func (rc *rollupConsumerImpl) extractAndVerifyRollups(br *common.BlockAndReceipts, blobs []*kzg4844.Blob) ([]*common.ExtRollup, error) {
 	rollups := make([]*common.ExtRollup, 0)
 	b := br.BlockHeader
 
-	for _, tx := range *br.RelevantTransactions() {
-		// go through all rollup transactions
+	for i, tx := range *br.RelevantTransactions() {
 		t := rc.MgmtContractLib.DecodeTx(tx)
 		if t == nil {
 			continue
 		}
 
-		rolTx, ok := t.(*ethadapter.L1RollupTx)
+		rollupHashes, ok := t.(*ethadapter.L1RollupHashes)
 		if !ok {
 			continue
 		}
 
-		r, err := common.DecodeRollup(rolTx.Rollup)
+		var blobHashes []gethcommon.Hash
+		var err error
+		if _, blobHashes, err = ethadapter.MakeSidecar(blobs); err != nil {
+			return nil, fmt.Errorf("could not create blob sidecar and blob hashes. Cause: %w", err)
+		}
+
+		if err := verifyBlobHashes(rollupHashes, blobHashes); err != nil {
+			rc.logger.Info(fmt.Sprintf("blob hashes in rollup at index %d do not match the rollup blob hashes", i))
+			continue
+		}
+
+		r, err := ethadapter.ReconstructRollup(blobs)
 		if err != nil {
-			rc.logger.Crit("could not decode rollup.", log.ErrKey, err)
-			return nil
+			return nil, fmt.Errorf("could not recreate rollup from blobs. Cause: %w", err)
```

Comment:
a bit inconsistent. Above, we "continue" on error. Here, we exit the function. We need to think what is the correct behaviour

Reply:
in the above case we want to continue because there might be more than one rollup in the block so we have to loop through them to see if the blob hashes we've received match any of the rollups. this is a critical error because the hashes match but we can't reconstruct the rollup. Have added some more comments to explain it better